docs/en/operations/settings/settings.md

```diff
@@ -1183,6 +1183,16 @@ See also:
 -   [insert_quorum_parallel](#settings-insert_quorum_parallel)
 -   [select_sequential_consistency](#settings-select_sequential_consistency)
 
+## majority_insert_quorum {#settings-majority_insert_quorum}
```

Comment:
Probably we use some value of `insert_quorum` instead of introducing new flag. But 0 already means `disabled`. Something like https://github.com/ClickHouse/ClickHouse/pull/39974 can be used, but not sure about how to upgrade setting type without breaking compatibility

Reply:
Right ,Earlier  I did think of using string for insert_quorum, but I did not use it because of upgrade issues, I discussed with @alesapin , and we thought that I can go ahead with using a new bool flag

Reply:
+1 for keeping it under the `insert_quorum` setting.
I'd prefer if both `all` and `majority` will be added.

Reply:
Okay if I understood it correctly , So something like `insert_quorum_type {numeric, majority, all}` , 
by default it will be numeric which will be value of insert_quorum.
Majority will be n/2 +1 , I think I should remove the max(majority, insert_quorum), It is complicated to documentation and if user wants it he can either use insert quorum or `all `option
And I guess a better name for `numeric` ?

Reply:
okay

Reply:
Okay if I understood it correctly , So something like` insert_quorum_type {numeric, majority, all}` ,
by default it will be numeric which will be value of `insert_quorum`.
Majority will be n/2 +1 , I think I should remove the `max(majority, insert_quorum)`, It is complicated to documentation and if user wants it he can either use insert quorum or all option
And I guess a better name for numeric ?
Or remove this setting all together, and add it into` insert_quorum`, Sorry I did not understand your comment

Reply:
I'd suggest to wait a bit until I finish https://github.com/ClickHouse/ClickHouse/pull/39974 and use `SET insert_quorum = 'auto'`

Reply:
okay

Reply:
@SachinSetiya fyi https://github.com/ClickHouse/ClickHouse/pull/39974 is merged

Reply:
Hi @vdimir , I am working on it, Will send it for review by today.